docs/proxy/datacenter_proxy/examples.md

```diff
@@ -0,0 +1,700 @@
+    ctx.verify_mode = ssl.CERT_NONE
+    httpHandler = request.HTTPSHandler(context=ctx)
+
+    opener = request.build_opener(httpHandler,proxy_handler)
+    return opener.open("https://api.apify.com/v2/browser-info").read()
+
+print(do_request())
+print("Should return a different clientIp than ")
+print(do_request())
+</marked-tab>
+
+
+<marked-tab header="PHP" lang="php">
+<?php
+function doRequest() {
+    $curl = curl_init("https://api.apify.com/v2/browser-info");
+    curl_setopt($curl, CURLOPT_RETURNTRANSFER, 1);
+    curl_setopt($curl, CURLOPT_PROXY, "http://proxy.apify.com:8000");
+    // Replace <YOUR_PROXY_PASSWORD> below with your password
+    // found at https://my.apify.com/proxy
+    curl_setopt($curl, CURLOPT_PROXYUSERPWD, "groups-SHADER+BUYPROXIES94952:<YOUR_PROXY_PASSWORD>");
+    $response = curl_exec($curl);
+    curl_close($curl);
+    return $response;
+}
+$response1 = doRequest();
+$response2 = doRequest();
+echo $response1;
+echo "\nShould have different clientIp than\n";
+echo $response2;
+?>
+</marked-tab>
+```
+
+## [](#using-the-apify-sdk) Using the Apify SDK
+
+If you're developing an actor using the [Apify SDK](https://sdk.apify.com), you can use Apify Proxy in:
+
+* [PuppeteerCrawler](https://sdk.apify.com/docs/api/puppeteer-crawler#docsNav) using the [createProxyConfiguration()](https://sdk.apify.com/docs/api/apify#apifycreateproxyconfigurationproxyconfigurationoptions) function.
+* [requestAsBrowser()](https://sdk.apify.com/docs/api/utils#utilsrequestasbrowseroptions) function by specifying proxy configuration in the options.
+* [launchPuppeteer()](https://sdk.apify.com/docs/typedefs/launch-puppeteer#docsNav) by specifying the configuration in the function's options.
+
+The Apify SDK's [ProxyConfiguration](https://sdk.apify.com/docs/api/proxy-configuration) enables you to choose which proxies you use for all connections. You can inspect the current proxy's URL and other attributes using the [ProxyInfo](https://sdk.apify.com/docs/typedefs/proxy-info) property in your crawler's [page function](https://sdk.apify.com/docs/typedefs/cheerio-crawler-options#handlepagefunction).
+
+### [](#rotate-ip-addresses) Rotate IP addresses
+
+IP addresses are selected at random from all available proxy servers.
+
+```marked-tabs
+<marked-tab header="PuppeteerCrawler" lang="javascript">
+const Apify = require("apify");
+
+Apify.main(async () => {
+    const requestList = await Apify.openRequestList(
+        "my-list", ["http://www.example.com"]
+    );
+    const proxyConfiguration = await Apify.createProxyConfiguration();
+
+    const crawler = new Apify.PuppeteerCrawler({
+        requestList,
+        proxyConfiguration,
+        handlePageFunction: async ({ page, request, proxyInfo }) => {
+            return Apify.pushData({ title: await page.title() });
+        },
+        handleFailedRequestFunction: ({ request }) => {
+            console.error("Request failed", request.url, request.errorMessages);
```

Comment:
The `handleFailedRequestFunction` is an optional function so better remove it from the examples.